src/chainstate/burn/distribution.rs

```diff
@@ -340,6 +442,217 @@ mod tests {
+            user_burns.clone(),
+            None,
+        );
+
+        assert_eq!(result.len(), 2, "Should be two miners");
+
+        result.sort_by_key(|sample| sample.candidate.txid);
+
+        assert_eq!(result[0].burns, 4);
+        assert_eq!(result[1].burns, 3);
+
+        // make sure that we're associating with the last commit in the window.
+        assert_eq!(result[0].candidate.txid, commits[5][0].txid);
+        assert_eq!(result[1].candidate.txid, commits[5][1].txid);
+
+        assert_eq!(result[0].user_burns.len(), 0);
+        assert_eq!(result[1].user_burns.len(), 1);
+
+        assert_eq!(result[1].user_burns[0].txid, user_burns[5][0].txid);
+
+        // test case 2:
+        //    miner 1:  4 4 5 4 5 3
+        //    miner 2:  4 4 4 4 4 1
+        //       ub  :  0 0 0 0 0 2
+        //               *split*
+
+        // miner 1 => min = 4, median = 4.
+        // miner 2 => min = 2, median = 4.
+
+        let commits = vec![
+            vec![
+                make_block_commit(4, 1, 1, 1, None, 1),
+                make_block_commit(4, 2, 2, 2, None, 1),
+            ],
+            vec![
+                make_block_commit(4, 3, 3, 3, Some(1), 2),
+                make_block_commit(4, 4, 4, 4, Some(2), 2),
+            ],
+            vec![
+                make_block_commit(5, 5, 5, 5, Some(3), 3),
+                make_block_commit(4, 6, 6, 6, Some(4), 3),
+            ],
+            vec![
+                make_block_commit(4, 7, 7, 7, Some(5), 4),
+                make_block_commit(4, 8, 8, 8, Some(6), 4),
+            ],
+            vec![
+                make_block_commit(5, 9, 9, 9, Some(7), 5),
+                make_block_commit(4, 10, 10, 10, Some(8), 5),
+            ],
+            vec![
+                make_block_commit(3, 11, 11, 11, Some(9), 6),
+                make_block_commit(1, 11, 11, 12, Some(10), 6),
+            ],
+        ];
+        let user_burns = vec![
+            vec![],
+            vec![],
+            vec![],
+            vec![],
+            vec![],
+            vec![make_user_burn(2, 11, 11, 1, 6)],
+        ];
+
+        let mut result = BurnSamplePoint::make_min_median_distribution(
+            commits.clone(),
+            user_burns.clone(),
+            None,
+        );
+
+        assert_eq!(result.len(), 2, "Should be two miners");
+
+        result.sort_by_key(|sample| sample.candidate.txid);
+
+        assert_eq!(result[0].burns, 4);
+        assert_eq!(result[1].burns, 3);
+
+        // make sure that we're associating with the last commit in the window.
+        assert_eq!(result[0].candidate.txid, commits[5][0].txid);
+        assert_eq!(result[1].candidate.txid, commits[5][1].txid);
+
+        assert_eq!(result[0].user_burns.len(), 1);
+        assert_eq!(result[1].user_burns.len(), 1);
+
+        assert_eq!(result[1].user_burns[0].txid, user_burns[5][0].txid);
+        assert_eq!(result[0].user_burns[0].txid, user_burns[5][0].txid);
+    }
+
```

Comment:
Can you add a test for the edge case where the PoX sunset takes effect?

Reply:
Yep, will add

Reply:
Added in 5c80770d4